ocaml/utils/language_extension_kernel.ml

```diff
@@ -0,0 +1,72 @@
+type maturity = Stable | Beta | Alpha
+
+(* Remember to update [all] when changing this type. *)
+type _ t =
+  | Comprehensions : unit t
+  | Local : unit t
+  | Include_functor : unit t
+  | Polymorphic_parameters : unit t
+  | Immutable_arrays : unit t
+  | Module_strengthening : unit t
+  | Layouts : maturity t
+
+type 'a language_extension_kernel = 'a t
+
+module Exist = struct
+  type t = Pack : _ language_extension_kernel -> t
+
+  let all =
+    [ Pack Comprehensions
+    ; Pack Local
+    ; Pack Include_functor
+    ; Pack Polymorphic_parameters
+    ; Pack Immutable_arrays
+    ; Pack Module_strengthening
+    ; Pack Layouts
+    ]
+end
+
+(* When you update this, update [of_string] below too. *)
+let to_string : type a. a t -> string = function
+  | Comprehensions -> "comprehensions"
+  | Local -> "local"
+  | Include_functor -> "include_functor"
+  | Polymorphic_parameters -> "polymorphic_parameters"
+  | Immutable_arrays -> "immutable_arrays"
+  | Module_strengthening -> "module_strengthening"
+  | Layouts -> "layouts"
+
+let of_string extn_name : Exist.t option =
+  match String.lowercase_ascii extn_name with
+  | "comprehensions" -> Some (Pack Comprehensions)
+  | "local" -> Some (Pack Local)
+  | "include_functor" -> Some (Pack Include_functor)
+  | "polymorphic_parameters" -> Some (Pack Polymorphic_parameters)
+  | "immutable_arrays" -> Some (Pack Immutable_arrays)
+  | "module_strengthening" -> Some (Pack Module_strengthening)
+  | "layouts" -> Some (Pack Layouts)
```

Comment:
I don't understand why `of_string` has stopped accepting `"layouts_{alpha,beta}"`

Reply:
`pair_of_string` still accepts `layouts_{alpha,beta}`. Previously, `of_string` was implemented in terms of `pair_of_string`; now, that dependency is reversed.

Reply:
Er, this isn't exactly right. One sec.

Reply:
It's certainly true that `of_string` now no longer accepts e.g. `layouts_alpha`. But that's fine&mdash;this function is used only to parse the third component of a Jane Syntax embedding. Maybe we should rename to `of_component`&mdash;WDYT?

Reply:
I think what I dislike is that "all the extension strings" are no longer in a uniform place, nor even a uniform file (right?).  I think it's valuable to keep that location centralized for editing.

Reply:
I moved them into the same file. The reason I didn't do this originally is that I dislike exposing `Exist_pair` from `Language_extension_kernel`, but I hear you that there's value in having all these strings in the same place. I think the new approach ends up being a bit nicer despite the trade-off.